include/tscpp/util/ts_errata.h

```diff
@@ -40,5 +40,16 @@ static constexpr std::array<swoc::TextView, 9> Severity_Names{
   {"Diag", "Debug", "Status", "Note", "Warn", "Error", "Fatal", "Alert", "Emergency"}
 };
 
+inline std::error_code
+ec_for()
+{
+  return {errno, std::system_category()};
```

Comment:
Is this right? The docs say that errno is in generic_category

https://en.cppreference.com/w/cpp/error/generic_category

Reply:
I considered that, but in practice `errno` is system specific (e.g., the value can vary between operating systems) and therefore, IMHO, should be `system_category`. If we look at the [documentation for system_category](https://en.cppreference.com/w/cpp/error/system_category) we see

"Obtains a reference to the static error category object for errors reported by the operating system", which is precisely what `errno` is.

Reply:
And that that text continues:
```
It is also required to override the virtual function [std::error_category::default_error_condition()](https://en.cppreference.com/w/cpp/error/error_category/default_error_condition) to map the error codes that match POSIX [errno](https://en.cppreference.com/w/cpp/error/errno) values to [std::generic_category](https://en.cppreference.com/w/cpp/error/generic_category).
```

So, this means that system category is required to forward errno values to generic category? TBH this error abstraction seems to provide more confusion that value 🤷